src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const newDateArray = [];
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = dateArray[fromFormat.indexOf('YY')];
+
+    dateArray[fromFormat.indexOf('YY')] = year < 30 ? '20' + year : '19' + year;
```

Comment:
The condition for converting a two-digit year to a four-digit year assumes that any year less than '30' should be prefixed with '20', and any year '30' or greater should be prefixed with '19'. This logic works for the given examples, but it's a rigid rule that may not apply in all contexts. Consider if this rule should be more flexible or configurable.